core/src/test/java/org/opensearch/sql/planner/physical/TrendlineOperatorTest.java

```diff
@@ -102,31 +101,101 @@ public void calculates_simple_moving_average_one_field_two_samples_three_rows()
                     AstDSL.computation(2, AstDSL.field("distance"), "distance_alias", SMA),
                     ExprCoreType.DOUBLE)));
 
-    plan.open();
-    assertTrue(plan.hasNext());
-    assertEquals(
-        ExprValueUtils.tupleValue(ImmutableMap.of("distance", 100, "time", 10)), plan.next());
-    assertTrue(plan.hasNext());
-    assertEquals(
-        ExprValueUtils.tupleValue(
-            ImmutableMap.of("distance", 200, "time", 10, "distance_alias", 150.0)),
-        plan.next());
-    assertTrue(plan.hasNext());
-    assertEquals(
-        ExprValueUtils.tupleValue(
-            ImmutableMap.of("distance", 200, "time", 10, "distance_alias", 200.0)),
-        plan.next());
-    assertFalse(plan.hasNext());
+    List<ExprValue> result = execute(plan);
+    assertEquals(3, result.size());
+    assertThat(
+        result,
+        containsInAnyOrder(
+            tupleValue(ImmutableMap.of("distance", 100, "time", 10)),
+            tupleValue(ImmutableMap.of("distance", 200, "time", 10, "distance_alias", 150.0)),
+            tupleValue(ImmutableMap.of("distance", 200, "time", 10, "distance_alias", 200.0))));
+  }
+
+  @Test
+  public void calculates_simple_moving_average_data_type_support_short() {
+    mockPlanWithData(
+        List.of(
+            tupleValue(ImmutableMap.of("distance", 100, "time", 10)),
+            tupleValue(ImmutableMap.of("distance", 200, "time", 10)),
+            tupleValue(ImmutableMap.of("distance", 200, "time", 10))));
+
+    var plan =
+        new TrendlineOperator(
+            inputPlan,
+            Collections.singletonList(
+                Pair.of(
+                    AstDSL.computation(2, AstDSL.field("distance"), "distance_alias", SMA),
+                    ExprCoreType.SHORT)));
```

Comment:
nit: seems like all these typed 

use `@ParameterizedTest` as an annotation on the test (instead of @Test)